src/internal/requests/tokenize.ts

```diff
@@ -20,7 +20,20 @@ export default (data: IDictionary) => {
   const errors = gateway.actions.validateData(data);
 
   if (errors.length > 0) {
-    return Promise.reject({ error: true, reasons: errors });
+    if (gateway.requiredSettings.indexOf("X-GP-Api-Key") !== -1) {
+      return Promise.reject({
+        error: {
+          code: "invalid_input",
+          detail: [{
+            data_path: "/card/card_number",
+            description: "Invalid data",
+          }],
+          message: "Invalid input data.",
+        },
+      });
```

Comment:
This should not be in this module and should exist in `src/internal/gateways/openedge/action-validate-data.ts`. That is the purpose of the abstracted data validation (so we don't have to recheck if a particular gateway is configured and run its custom code if present).